src/keyboard_listener.h

```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+#include <string>
+
+class KeyboardListener{
+public:
+    virtual void notify_keyboard_key_pressed(std::string keypress) = 0;
```

Comment:
Maybe it's just me, but I feel like just `keyboard_key_pressed` seems more intuitive to read, or perhaps `handle_keyboard_key_pressed`. With the `notify` at the front, it reads like the listener is committing the action of doing the notifying, when really it's being notified.

I think this issue becomes really apparent in lines 38 and 39 of `keyboard.h`

```c++
private:
     void notify_events(SDL_Event e);
     void notify_key(std::string key); 
};
```

Even though they read the same, `notify_key` is notifying the listeners, while `notify_events` is handling incoming events as a listener.